### PR TITLE
Assign facilitator control

### LINKED ISSUE #3178: Meeting sidebar facilitator control implemented
Based on design efforts from #3077

### Acceptance Criteria
- Precedes #3180 
- The sidebar has a dropdown control that switches the facilitator
- The control shows after the name of the meeting, consider yanking the first label once the meeting name can be changed
- Here’s [an early iteration](https://www.figma.com/file/0lOTlvsj87GBFFzgee9OkVWi/Meeting-Layout?node-id=358%3A12701) of the UI that can be improved (here are the [latest designs](https://www.figma.com/file/0lOTlvsj87GBFFzgee9OkVWi?node-id=479%3A80963&utm_campaign=team&utm_source=slack&utm_medium=message&utm_content=file_comment#11592859))
- The default state shows the current facilitator: avatar, preferred name, dropdown icon
- The menu shows all team members (avatar, preferred name), even if not connected, because async UX. Exclude the current facilitator from the list. Blurring the menu is the way to back out if the user doesn’t want to pick a new facilitator.
- The interaction is like changing the owner of a task card, whoever gets selected is the new facilitator, refactor default facilitator logic as needed
- Removes affordance from avatar menu only if necessary, otherwise removed in #3180
- The facilitator control doesn’t show in the lobby
- The facilitator control is read-only in the demo
- The facilitator control is read-only for a team with 1 member

### Estimated Effort
- 11 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

### PR DESCRIPTION
**Pull request by ackernaut**
Fixes #3178
### Tests
The assign facilitator control:
- [ ] Doesn’t show in lobby
- [ ] Is read-only in the demo
- [ ] Is read-only when the team only has 1 member
- [ ] Tapping reveals a menu
- [ ] The menu shows all team members except the facilitator
- [ ] Tapping a team member promotes them to facilitator (showing the snackbar alert)
- [ ] Clicking outside of the menu (instead of on a team member) closes the menu
- [ ] The menu updates to show new teammates as they join during the meeting

### Note
- Latest styles for this in #3228

### CODE REVIEW

File: packages/client/components/AssignFacilitatorMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          />
+        )
+      })}
+    </Menu>
+  )
+}
+
+export default createFragmentContainer(withAtmosphere(AssignFacilitatorMenu), {
+  viewer: graphql`
+    fragment AssignFacilitatorMenu_viewer on User {
```
Comment (mattkrick) [inline packages/client/components/AssignFacilitatorMenu.tsx]: the only thing you pass in is a team, which is already populated by the parent object. that means you get rid of the `Root` component

File: packages/client/components/AssignFacilitatorMenuRoot.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React from 'react'
```
Comment (mattkrick) [inline packages/client/components/AssignFacilitatorMenuRoot.tsx]: remove this component & just pass in `team`

File: packages/client/components/AssignFacilitator.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      {menuPortal(<AssignFacilitatorMenuRoot menuProps={menuProps} team={team} newMeeting={newMeeting} />)}
+    </AssignFacilitatorBlock>
+  )
+}
+
+export default createFragmentContainer(withAtmosphere(AssignFacilitator), {
+  team: graphql`
+    fragment AssignFacilitator_team on Team {
+      ...AssignFacilitatorMenuRoot_team
+      teamMembers(sortBy: "checkInOrder") {
```
[the remark this answers is not in the document]
Comment (ackernaut) [reply packages/client/components/AssignFacilitator.tsx]: I’m using team to get length and make the control read-only if it’s a team of 1

**Comment by ackernaut** [comment]
@mattkrick thanks for the initial review! I made the suggested changes, so ready for one more pass. The typecheck fail looks like something related to build and not the components?

**Comment by mattkrick** [comment]
yep! @ackernaut we now run a typecheck in push, which is a nice way to make sure we don't let typescript errors slip in without increasing time spent during the build.

File: packages/client/components/AssignFacilitatorMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+const AssignFacilitatorMenu = (props: Props) => {
+  const {menuProps, team} = props
+  const {newMeeting} = team
+  const {facilitatorUserId, id: meetingId} = newMeeting!
+  const {teamMembers} = team || {teamMembers: []}
+  const assignees = useMemo(
+    () => teamMembers
+      .filter((teamMember) => teamMember.userId !== facilitatorUserId)
+      .sort((a,b) => (a.preferredName > b.preferredName) ? 1 : ((b.preferredName > a.preferredName) ? -1 : 0)),
```
Comment (mattkrick) [inline packages/client/components/AssignFacilitatorMenu.tsx]: well now i'm curious. does this have any advantage over the following? 
`sort((a,b) => a.prefferedName < b.preferredName ? -1 : 1)`

Comment (ackernaut) [reply packages/client/components/AssignFacilitatorMenu.tsx]: I copied from SO, and yeah the original looks verbose. I’ll try this minimal version and verify.

**Pull request merged**